### PR TITLE
💀🔥 Remove Forum rolling update and some methods

### PR DESCRIPTION
**Pull request by josemigallas**
Remove the rolling update and some methods that always returns false as a consequence.

Related: [THREESCALE-6714](https://issues.redhat.com/browse/THREESCALE-6714)

### CODE REVIEW

**Review by mayorova — APPROVED** [review]
I understand that you're going step by step, and this is the 1st one. Looks good to me.

File: test/integration/cms/builtin_static_pages_test.rb [hunk 1]
```diff
@@ -12,19 +13,5 @@ def setup
...
-      @simple_layout = SimpleLayout.new(@provider)
-      @simple_layout.import!
-
-      @layout = @provider.layouts.create(system_name: 'custom_layout')
-      static_page = @provider.builtin_static_pages.find_by_system_name!('forum/forums/show')
-      static_page.update_attribute(:layout, @layout)
-
-      get '/forum'
-      assert_template 'layouts/custom_layout'
-    end
```
Comment (jlledom) [inline test/integration/cms/builtin_static_pages_test.rb:26]: It would be good to figure out a way to keep this test even after removing all static pages

File: test/unit/abilities/provider_admin_test.rb [hunk 2]
```diff
@@ -11,9 +11,6 @@ def setup
 
     def test_forum
       Account.any_instance.expects(:provider_can_use?).returns(true).at_least_once
```
Comment (jlledom) [inline test/unit/abilities/provider_admin_test.rb:13]: You don't need this mock anymore

File: test/unit/forms/onboarding/api_form_test.rb
Comment (jlledom) [inline test/unit/forms/onboarding/api_form_test.rb:1]: This, and other tests under `/test/unit/forms` doesn't seem related to forums. How are they related?

File: test/unit/simple_layout_test.rb [hunk 3]
```diff
@@ -13,7 +13,7 @@ def setup
 
     assert SimpleLayout.new(@provider).import!
     assert_equal 2, @provider.layouts.count
-    assert_equal 14, @provider.builtin_static_pages.count
+    assert_equal 0, @provider.builtin_static_pages.count
```
Comment (jlledom) [inline test/unit/simple_layout_test.rb:16]: This test doesn't make sense anymore after this change. Why not mocking a static page and ensure it's imported?

**Comment by jlledom** [comment]
If tests pass, I'd say the code is safe enough to merge. But I wonder what happens with customers already having forums enabled.

File: app/controllers/sites/forums_controller.rb [hunk 4]
```diff
@@ -2,7 +2,7 @@ class Sites::ForumsController < Sites::BaseController
   activate_menu :audience, :forum, :settings
 
   before_action :authorize_forum_feature, :find_settings
-  before_action :active_settings_menu, unless: :forum_enabled?
```
Comment (mayorova) [inline app/controllers/sites/forums_controller.rb:5]: This `before_action` should be removed completely, because it activates the menu `:audience, :cms, :forum_settings` that is already removed here https://github.com/3scale/porta/pull/3602/files#diff-ac12a034e5e76892ef8e4d01b6b64340355bddc726088fc858bd45517dd30d54L170

Well, the whole controller doesn't make sense after forum removal, but it's OK to go step by step :grimacing:

File: app/lib/simple_layout.rb [hunk 5]
```diff
@@ -308,9 +308,8 @@ def setup_main_layout!
 
   private
 
-  def forum_builtin_static_pages
-    return [] unless provider.provider_can_use?(:forum)
-
+  def forum_builtin_static_pages # rubocop:disable Metrics/MethodLength
+    # TODO: clean up
```
Comment (mayorova) [inline app/lib/simple_layout.rb:312]: Why leave this method?
Just assuming that this is left as a reference?

File: test/integration/cms/builtin_static_pages_test.rb (hunk 1 above)
Comment (mayorova) [reply test/integration/cms/builtin_static_pages_test.rb:26]: I tried to come up with one, but it's a bit tricky...

I am wondering whether we actually need it, because it seems that Builtin Static Pages are only used for Forum-related pages.

Comment (josemigallas) [reply test/integration/cms/builtin_static_pages_test.rb:26]: I agree, I think it was meant for Forum pages.

File: test/unit/forms/onboarding/api_form_test.rb
Comment (josemigallas) [reply test/unit/forms/onboarding/api_form_test.rb:1]: You're right, I mixed up "form" with "forum" hehe

File: app/lib/simple_layout.rb (hunk 5 above)
Comment (josemigallas) [reply app/lib/simple_layout.rb:312]: I will remove "builtin pages" stuff later (methods, events, tests, etc)

File: app/controllers/provider/admin/cms/builtin_static_pages_controller.rb [hunk 6]
```diff
@@ -3,7 +3,9 @@ class Provider::Admin::CMS::BuiltinStaticPagesController < Provider::Admin::CMS:
   private
 
   def templates
+    # Return 404 when trying to edit all existing forum templates. TODO: Remove forums THREESCALE-6714
     current_account.builtin_static_pages
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply app/controllers/provider/admin/cms/builtin_static_pages_controller.rb:7]: Are we 100% positive there won't ever be any other static builtin pages? Even if so, I rather remove the whole controller all at once (in then next PR)

File: test/unit/simple_layout_test.rb (hunk 3 above)
Comment (josemigallas) [reply test/unit/simple_layout_test.rb]: I've been thinking about this but here's the thing: the test is about `SimpleLayout` importing the forum templates, not testing the method itself. It's not really a unit test but an integration test of sorts. Looking at the other tests they all test against hard data, nothing is mocked.

So we could mock that SimpleLayout has some builtin static templates but I don't think that's what this test is for.

**Review by lvillen — APPROVED** [review]
